components/studio/bin/hab-studio.sh

```diff
@@ -909,6 +909,12 @@ exit_with() {
   exit $2
 }
 
+# **Internal** Builds up a secret environmt based on the prefix `HAB_STUDIO_SECRET_`
+# to pass into the studio
+load_secrets() {
+  echo $($bb env | $bb awk -F '=' '/^HAB_STUDIO_SECRET_/ {gsub(/HAB_STUDIO_SECRET_/, "", $0); print}')
+}
```

Comment:
I think we should look for a very small subset of variables that could override critical behavior inside a Studio. I entered a vanilla Studio with `env HAB_STUDIO_SUP=false hab studio enter`, ran `env | awk -F '=' '{print $1}' | sort` to get the list of exported environment variables, and from that would think we want to be careful about these ones:

* `HAB_BINLINK_DIR` - set by the Studio code and affects where programs are binlinked to show up on `PATH`
* `HAB_ORIGIN` - set by the Studio code and consumed by the `build` program and which could affect which keys are loaded into the Studio
* `HOME` - set to a default `/root` which could be used by some autoconf-style tooling in packages. Would also be used by some tools when computing default caching/config directories
* `LC_ALL` - set by the Studio code to `POSIX`, this affects the compiled locales of certain packages. If this needs to be set specific for a package, this should go in a `plan.sh` to override
* `PATH` - set by the Studio code, specifically the Studio type which pulls in full paths to certain Habitat packages. If a `PATH` needs to be augmented for building, that should go in a `plan.sh` and if needed in an interactive Studio, that should go in a `.studiorc`
* `PWD` - set by the Studio code to `/src` as the initial directory
* `STUDIO_TYPE` - set by the Studio code and can control the behavior of certain base packages. This should be treated as an environmental invariant (i.e. does not change)
* `TERM` - set by the Studio code and can control progress bars, interactive modes, coloring ,etc
* `TERMINFO` - set by the Studio code (depending on Studio type) and will refer to a `terminfo` database in a particular Habitat package

That would mean that, given the above list, `HAB_STUDIO_SECRET_TERMINFO` wouldn't be allowed to enter the Studio as `TERMINFO`. I suspect a warning to a user is sufficient (rather than a hard-fail), but worth a look.